pkg/istio-agent/xds_proxy.go

```diff
@@ -492,7 +498,7 @@ func (p *XdsProxy) rewriteAndForward(con *ProxyConnection, resp *discovery.Disco
 }
 
 func forwardToEnvoy(con *ProxyConnection, resp *discovery.DiscoveryResponse) {
-	if !v3.IsEnvoyType(resp.TypeUrl) {
+	if !v3.IsEnvoyType(resp.TypeUrl) && !strings.HasPrefix(resp.TypeUrl, "istio.io/debug/") {
```

Comment:
why do we forward debug types to envoy? or is it not "envoy" but the "tap client" and the comments are misleading?

Reply:
The function name is misleading.  Before I added this line, the response from Istiod was dropped.

Reply:
For added security, we could change the logic to only allow istio.io/debug to forward to the tap clients, and only allow Envoy types to forward to Envoy.

Reply:
OK, I built a version and confirmed your suspicion.  This code doesn't work.  Looking deeper...